crates/ruff_server/src/trace.rs

```diff
@@ -0,0 +1,151 @@
+use lsp_types::{notification::Notification, TraceValue};
+use serde::Deserialize;
+use std::sync::{Mutex, OnceLock};
+use tracing::{level_filters::LevelFilter, Event};
+use tracing_subscriber::{layer::SubscriberExt, Layer};
+
+use crate::server::ClientSender;
+
+static LOGGING_SENDER: OnceLock<ClientSender> = OnceLock::new();
+
+static TRACE_VALUE: Mutex<lsp_types::TraceValue> = Mutex::new(lsp_types::TraceValue::Off);
+
+pub(crate) fn set_trace_value(trace_value: TraceValue) {
+    let mut global_trace_value = TRACE_VALUE
+        .lock()
+        .expect("trace value mutex should be available");
+    *global_trace_value = trace_value;
+}
```

Comment:
The PR made a note that this would be used in the future:

> This PR also introduces a $/setTrace handler, which will be used in the future by our tracing span subscriber.

I wanted to at least add a handler for this so that we don't get a warning about a nonexistent handler every time the trace level changes.